src/types.d.ts

```diff
@@ -481,31 +488,23 @@ export interface Chunk extends ChunkMetadata, ChunkData {
   tx_path: Buffer;
 }
 
+export interface ChunkDataByAnySourceParams {
```

Comment:
I'm thinking we should make this a union type of some sort. We either need absolute offset and size or size, data root, and relative offset. That could be something we revisit at the end though when we're putting all the pieces together.